hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -244,7 +244,9 @@ private void resetFileGroupsReplaced(HoodieTimeline timeline) {
       }
     });
 
-    Map<HoodieFileGroupId, HoodieInstant> replacedFileGroups = resultStream.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    // Duplicate key error when insert_overwrite same partition in multi writer, distinct HoodieFileGroupId,If repeated, save instant with the larger timestamp
+    Map<HoodieFileGroupId, HoodieInstant> replacedFileGroups = resultStream.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue,
+            (instance1, instance2) -> instance1.compareTo(instance2) < 0 ? instance2 : instance1));
     resetReplacedFileGroups(replacedFileGroups);
```

Comment:
It is better to use `HoodieTimeline.compareTimestamps` for comparing the instants..